src/core/view/overlays/ShapeToolView.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Xournal++
+ *
+ * View active shape tools (rectangle, ellipse and so on)
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+#pragma once
+
+#include <memory>
+
+#include "util/DispatchPool.h"
+#include "view/Repaintable.h"
+
+#include "OverlayView.h"
+
+class BaseShapeHandler;
+class Range;
+class Stroke;
+
+namespace xoj::view {
+struct Context;
+class StrokeView;
+
+class ShapeToolView: public OverlayView, public Repaintable, public xoj::util::Listener<ShapeToolView> {
+
+public:
+    ShapeToolView(BaseShapeHandler* toolHandler, const Repaintable* parent);
+    virtual ~ShapeToolView() noexcept;
+
+    /**
+     * @brief Draws the overlay to the given context
+     */
+    void draw(const Context& ctx) const override;
+
+    bool isViewOf(const OverlayBase* overlay) const override;
+
+    /**
+     * Listener interface
+     */
+    static constexpr struct PAINT_REQUEST_t {
```

Comment:
Please use snake_case or PascalCase for types. `snake_case` for STL like/mimic structures. `PascalCase` for Xournal internal classes.